the "udpate_maxim" branch, to merge

buddhasystem opened the pull request:
1. A small addition to the Observation ctor signature, which can now take a tuple of (begin, end) for start and end times, for convenience of plugging into other components. Some general code improvements.
2. Fixed discrepancies in the satellite test notebook (API had evolved a while back)
3. Added a stub for the "Simple Satellite" class, to reduce clutter in simple applications. This will help to easily create unified files for the Sun and two satellites sky coordinates, for the OpSim simulations.

To be done still -- conversion in both directions, from astropy Time to MJD string and back, to more easily handle the Satellite. Work in progress. 

The current version of OpSim has dependencies on this branch, so if all is well it would be not a bad idea to merge even though the changes are not dramatic.

buddhasystem commented on lusee/Observation.py:
```diff
@@ -4,98 +4,111 @@
-        lun_lat_deg     =   -23.814,
-        lun_long_deg    =   182.258,
-        lun_height_m    =   0,
+        time            =   default_time,
```
> Why do we call this time? Time in my mind is a point in time, not a range (like lunar_day, or hour etc). At least call it `time_range`.

Sure, I will change that. To me "lunar_day" was also like an integer. But it's easy to change anyhow.

buddhasystem commented on lusee/Observation.py:
```diff
@@ -4,98 +4,111 @@
+        self.time_start = Time(tpl[0])
+        self.time_end   = Time(tpl[1])
+    
+    @classmethod
```
Parsing the string " to " in the time specification seemed not optimal. With that small addition, it's easier to pass strings for "start" and "end", and also with a small change morph this argument to astropy Time class. Just seemed more flexible.

slosar replied:
Sorry, I didn't see you are actually setting -- from cursory reading I didn't see the return statement.

buddhasystem commented on lusee/Satellite.py:
```diff
@@ -154,4 +156,20 @@ def get_track_coverage(self, Nphi=10, Nmu=10):
     
+##############################################
+class SimpleSatellite:
+    fctr = 1./(180.0*np.pi)
```
I see. Probably I made a mistake with this "factor", will fix.

slosar commented on lusee/Observation.py:
```diff
@@ -4,98 +4,111 @@
 
+
+    # ---
+    def time_from_range(self, tpl):
```
```suggestion
    def set_time_from_range(self, tpl):
```
Perhaps this?

buddhasystem replied:
OK, will do.

buddhasystem commented on lusee/Satellite.py:
```diff
@@ -154,4 +156,20 @@ def get_track_coverage(self, Nphi=10, Nmu=10):
+class SimpleSatellite:
+    fctr = 1./(180.0*np.pi)
+    """
+    Simplified Satellite class with only basic coordinates
```
I'll refactor. I wanted something readable for the time being, we'll need to design a better way to define satellites, it would seem.

buddhasystem commented:
I've done more changes, renamed the time range, removed the "Simple Satellite", did cleanup and updated the satellite notebook. I'll reopen the PR for an easier review.
